File: BiosenseSignal_Flutter_SDK_5.6.3/android/src/main/kotlin/com/biosensesignal/flutter_sdk/BiosenseSignalPreviewFactory.kt
package com.biosensesignal.flutter_sdk

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

object BiosenseSignalPreviewFactory: PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    const val cameraPreviewId = "plugins.biosensesignal.com/camera_preview_view"
    private var cameraPreview: BiosenseSignalCameraPreview? = null
    private var imageDataSource: ImageDataSource? = null

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return BiosenseSignalCameraPreview(context).also {
            cameraPreview = it
            cameraPreview?.setDataSource(imageDataSource ?: return@also)
        }
    }

    fun setDataSource(imageDataSource: ImageDataSource) {
        this.imageDataSource = imageDataSource
        cameraPreview?.setDataSource(imageDataSource)
    }
}

File: BiosenseSignal_Flutter_SDK_5.6.3/ios/Classes/BiosenseSignalPreviewFactory.swift
//
//  BiosenseSignalPreviewFactory.swift
//  biosensesignal_flutter_sdk
//
//  Created on 26/03/2023.
//

import Foundation


class BiosenseSignalPreviewFactory: NSObject, FlutterPlatformViewFactory {
    
    static let shared: BiosenseSignalPreviewFactory = BiosenseSignalPreviewFactory()
    static let cameraPreviewId = "plugins.biosensesignal.com/camera_preview_view"
    
    private var cameraPreview: BiosenseSignalCameraPreview?
    private var imageDataSource: ImageDataSource?
    
    private override init() { }
    
    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        let preview = BiosenseSignalCameraPreview()
        preview.setDataSource(imageDataSource: imageDataSource)
        return preview
    }
    
    func setDataSource(imageDataSource: ImageDataSource) {
        self.imageDataSource = imageDataSource
        cameraPreview?.setDataSource(imageDataSource: imageDataSource)
    }
    
    
    
}
